tests/SprykerSdkTest/Spryk/Model/Spryk/Filter/TypedArrayFilterTest.php

```diff
@@ -0,0 +1,115 @@
+ * @group SprykerSdkTest
+ * @group Spryk
+ * @group Model
+ * @group Spryk
+ * @group Filter
+ * @group TypedArrayFilterTest
+ * Add your own group annotations below this line
+ */
+class TypedArrayFilterTest extends Unit
+{
+    /**
+     * @return void
+     */
+    public function testTypedArrayShouldConvertTypedArrayToArray(): void
+    {
+        $inputParameters = 'string[] $foo';
+        $expectedResult = 'array $foo';
+
+        $this->assertFilterResult($inputParameters, $expectedResult);
+    }
+
+    /**
+     * @return void
+     */
+    public function testTypedArrayShouldConvertTypedArraysToArrays(): void
+    {
+        $inputParameters = 'string[] $foo, int[] $bar';
+        $expectedResult = 'array $foo, array $bar';
+
+        $this->assertFilterResult($inputParameters, $expectedResult);
+    }
+
+    /**
+     * @return void
+     */
+    public function testTypedArrayShouldConvertTypedArrayToArrayWithMixedValues(): void
+    {
+        $inputParameters = 'string[] $foo, int $bar';
+        $expectedResult = 'array $foo, int $bar';
+
+        $this->assertFilterResult($inputParameters, $expectedResult);
+    }
+
+    /**
+     * @return void
+     */
+    public function testTypedArrayShouldConvertTypedArrayToArrayWithMultipleMixedValues(): void
+    {
+        $inputParameters = 'string[] $foo, int $bar, float[] $baz, iterable $qux';
+        $expectedResult = 'array $foo, int $bar, array $baz, iterable $qux';
+
+        $this->assertFilterResult($inputParameters, $expectedResult);
+    }
+
+    /**
+     * @return void
+     */
+    public function testTypedArrayShouldConvertClassTypedArrayToArrayWithMultipleValues(): void
+    {
+        $inputParameters = '\stdClass[] $foo, \DateTime $bar, array $baz, \Spryker\Glue\GlueApplication\Rest\JsonApi\RestResourceInterface[] $qux';
+        $expectedResult = 'array $foo, \DateTime $bar, array $baz, array $qux';
+
+        $this->assertFilterResult($inputParameters, $expectedResult);
+    }
+
+    /**
+     * @return void
+     */
+    public function testTypedArrayShouldConvertTypedArrayToArrayForReturnTypeHint(): void
+    {
+        $inputParameters = '\stdClass[]';
+        $expectedResult = 'array';
+
+        $this->assertFilterResult($inputParameters, $expectedResult);
+    }
+
+    /**
+     * @return void
+     */
+    public function testTypedArrayShouldNotConvertTypeHintWhenItsNotTypedArray(): void
+    {
+        $inputParameters = '\stdClass';
+        $expectedResult = '\stdClass';
+
+        $this->assertFilterResult($inputParameters, $expectedResult);
+    }
+
+    /**
+     * @param string $inputParameters
+     * @param string $expectedResult
+     *
+     * @return void
+     */
+    protected function assertFilterResult(string $inputParameters, string $expectedResult): void
+    {
+        $filter = new TypedArrayFilter();
+        $actualResult = $filter->filter($inputParameters);
+
+        $this->assertSame($expectedResult, $actualResult);
+    }
+}
```

Comment:
You need to add tests (and support for in the filter) for default value passed as the method parameter: `'?\stdClass'`, `'string $foo = null'`, `'string[] $foo = null'`, `'/Class/Bar[] $foo = null'`...

Reply:
Done